bin/prettier.js

```diff
@@ -361,30 +360,51 @@ function writeOutput(result) {
 }
 
 function eachFilename(patterns, callback) {
-  patterns.forEach(pattern => {
-    if (!glob.hasMagic(pattern)) {
-      if (shouldIgnorePattern(pattern)) {
-        return;
-      }
-      callback(pattern);
-      return;
-    }
+  if (ignoreNodeModules) {
+    patterns = patterns.concat(
+      ignoreNodeModulesGlobs.map(pattern => "!" + pattern)
+    );
+  }
 
-    glob(pattern, globOptions, (err, filenames) => {
-      if (err) {
-        console.error("Unable to expand glob pattern: " + pattern + "\n" + err);
-        // Don't exit the process if one pattern failed
-        process.exitCode = 2;
-        return;
+  return globby(patterns, globOptions)
+    .then(filePaths => {
+      if (filePaths.length === 0) {
+        const message =
+          (files => {
+            // Seperate files into last (last file) and initial - all the others
+            const initial = files.slice(0);
+            const last = initial.pop();
+
+            // Join into a comma seperated string of file names
+            const ending = files.length > 1
+              ? "and " + last + " do"
+              : last + " does";
+            return (initial.join(", ") + " " + ending).trim();
+          })(patterns) + " not match any files";
```

Comment:
It took me a couple of readings to understand what was going on here. Could you extract this into a `humanList(array)` function or something? While reading this code I tested stylelint and learned that it formats the error this way, but does this formatting help so much? I'd be fine with something like:

```
No matching files. Patterns tried:
${patterns.join(' ')}
```

I don't know...